packages/smithy/smithy_aws/test/http/aws_retryer_test.dart

```diff
@@ -0,0 +1,379 @@
+  final TestSuiteGiven given;
+  final List<TestCase> responses;
+}
+
+@_testCaseSerializable
+class TestSuiteGiven {
+  const TestSuiteGiven({
+    required this.maxAttempts,
+    required this.initialRetryTokens,
+    required this.exponentialBase,
+    required this.exponentialPower,
+    required this.maxBackoffTime,
+  });
+
+  factory TestSuiteGiven.fromJson(Map json) => _$TestSuiteGivenFromJson(json);
+
+  final int maxAttempts;
+  final int initialRetryTokens;
+  final double exponentialBase;
+  final double exponentialPower;
+  final int maxBackoffTime;
+
+  @override
+  String toString() => [
+        maxAttempts,
+        initialRetryTokens,
+        exponentialBase,
+        exponentialPower,
+        maxBackoffTime
+      ].join('|');
+}
+
+@_testCaseSerializable
+class TestCase {
+  const TestCase({
+    required this.response,
+    required this.expected,
+  });
+
+  factory TestCase.fromJson(Map json) => _$TestCaseFromJson(json);
+
+  final TestCaseResponse response;
+  final TestCaseExpected expected;
+}
+
+@_testCaseSerializable
+class TestCaseResponse {
+  const TestCaseResponse({
+    required this.statusCode,
+  });
+
+  factory TestCaseResponse.fromJson(Map json) =>
+      _$TestCaseResponseFromJson(json);
+
+  final int statusCode;
+}
+
+@_testCaseSerializable
+class TestCaseExpected {
+  const TestCaseExpected({
+    required this.outcome,
+    required this.retryQuota,
+    this.delay,
+  });
+
+  factory TestCaseExpected.fromJson(Map json) =>
+      _$TestCaseExpectedFromJson(json);
+
+  final Outcome outcome;
+  final int retryQuota;
+  final int? delay;
+}
+
+List<TestSuite> loadTestSuite(String testSuiteYaml) {
+  final yaml = loadYamlStream(testSuiteYaml);
+  return yaml.cast<Map>().map(TestSuite.fromJson).toList();
+}
+
+void main() {
+  final standardTestSuite = loadTestSuite(_standardTestCases);
+  group('AWSRetryer', () {
+    group('Standard Mode', () {
+      for (final testSuite in standardTestSuite) {
+        final maxAttempts = testSuite.given.maxAttempts;
+        final retryer = AWSRetryer(
+          initialRetryTokens: testSuite.given.initialRetryTokens,
+          exponentialBase: testSuite.given.exponentialBase,
+          exponentialPower: testSuite.given.exponentialPower,
+          maxBackoffTime: Duration(seconds: testSuite.given.maxBackoffTime),
+        );
+        test('${testSuite.given}', () {
+          runZoned(() async {
+            final outcome = testSuite.responses.last;
+            final Matcher expectation;
+            switch (outcome.expected.outcome) {
+              case Outcome.success:
+              case Outcome.retryRequest:
+                expectation = completes;
+                break;
+              case Outcome.maxAttemptsExceeded:
+              case Outcome.retryQuotaExceeded:
+                expectation = throwsA(isA<_TransientSmithyException>());
+                break;
+            }
+
+            // TODO(dnys1): Try to get fake_async to work properly to avoid
```

Comment:
Whats the unnecessary delay?

Reply:
There's a `Future.delayed` inside the retryer